pkg/cmd/server/origin/master.go

```diff
@@ -164,6 +164,8 @@ func (c *MasterConfig) Run(protected []APIInstaller, unprotected []APIInstaller)
 	handler := c.versionSkewFilter(safe)
 	handler = c.authorizationFilter(handler)
 	handler = c.impersonationFilter(handler)
+	// audit handler must comes before the impersonationFilter to read the original user
+	handler = c.auditHandler(handler)
```

Comment:
@deads2k that surprises me, I thought the handler order is FIFO, but from what I've tested it's LIFO.